src/audioin.js

```diff
@@ -224,25 +225,43 @@ define(function (require) {
     }
   };
 
-  /**
-   *  Returns a list of available input sources. Some browsers
-   *  give the client the option to set their own media source.
-   *  Others allow JavaScript to determine which source,
-   *  and for this we have listSources() and setSource().<br/>
-   *
-   *  @method  listSources
-   *  @return {Array}
-   */
   p5.AudioIn.prototype.listSources = function() {
+    console.log('listSources is deprecated - please use AudioIn.getSources')
     console.log('input sources: ');
     console.log(p5sound.inputSources);
     if (p5sound.inputSources.length > 0) {
-      return p5sound.inputSources;
+      //return p5sound.inputSources;
     } else {
       return 'This browser does not support MediaStreamTrack.getSources()';
     }
   };
-
+  /**
+   * Returns a list of available input sources. Chrome allows the 
+   * the user to access a list of possible media sources. Firefox 
+   * allows the user to choose from input sources in the permissions
+   * dialogue.
+   *
+   * @method  getSources
+   * @param  {Function} callback [a callback to handle the sources 
+   * when they have been enumerated]
+   * @return {[type]}            [description]
+   */
```

Comment:
@polyrhythmatic just a small note about documentation.

Param and return are formatted like this:

```{type} name description```

I found it a little confusing that docblockr puts  ``[ ]`` around type and description, because we don't actually want any square brackets there. The only place we use square brackets is around the name, and only to specify that it's an optional parameter. So if the callback here was optional, we could say 

```
  * @param  {Function} [callback] a callback to handle the sources 
  *                               when they have been enumerated
```

I'll update this for now. More info on p5 documentation [here](https://github.com/processing/p5.js/wiki/Inline-documentation#specify-parameters)